nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/StopDomainCommand.java

```diff
@@ -190,14 +190,21 @@ protected void doCommand() throws CommandException {
 
     /**
      * Wait for the server to die.
+     *
+     * @param watchedPid
      */
-    private void waitForDeath() throws CommandException {
+    private void waitForDeath(File watchedPid) throws CommandException {
         if (!programOpts.isTerse()) {
             // use stdout because logger always appends a newline
             System.out.print(Strings.get("StopDomain.WaitDASDeath") + " ");
         }
         final Duration timeout = Duration.ofMillis(DEATH_TIMEOUT_MS);
-        final Supplier<Boolean> deathSign = () -> !ProcessUtils.isListening(addr);
+        final Supplier<Boolean> deathSign;
+        if (isLocal()) {
```

Comment:
To avoid https://github.com/eclipse-ee4j/glassfish/pull/24246/files#r1072881827 - could `watchedPid` be checked here against `null` instead?

Reply:
On the other hand `isLocal` has more meaning than `watchedPid != null`.